deploy/template.yaml

```diff
@@ -1325,6 +1325,99 @@ Resources:
               Bool:
                 "aws:SecureTransport": true
 
+  #
+  # DynamoDB
+  #
+
+  SessionsDynamoDB:
+    Type: AWS::DynamoDB::Table
+    Properties:
+      AttributeDefinitions:
+        - AttributeName: "id"
+          AttributeType: "S"
+        - AttributeName: "subjectId"
+          AttributeType: "S"
+      BillingMode: PAY_PER_REQUEST
+      ContributorInsightsSpecification:
+        Enabled: true
+      GlobalSecondaryIndexes:
+        - IndexName: "subjectIdIndex"
+          KeySchema:
+            - AttributeName: "subjectId"
+              KeyType: "HASH"
+          Projection:
+            ProjectionType: KEYS_ONLY
+      KeySchema:
+        - AttributeName: "id"
+          KeyType: "HASH"
+      SSESpecification:
+        KMSMasterKeyId: !Ref DynamoDBKmsKey
+        SSEEnabled: true
+        SSEType: KMS
+      TableClass: STANDARD
+      TableName: !Sub ${AWS::StackName}-SessionsDynamoDB
+      Tags:
+        - Key: Product
+          Value: !Ref ProductTagValue
+        - Key: System
+          Value: Accounts
+        - Key: Environment
+          Value: !Ref Environment
+        - Key: Owner
+          Value: !Ref OwnerTagValue
+        - Key: Source
+          Value: !Ref SourceTagValue
+        - Key: Application
+          Value: Shared
+
+  DynamoDBKmsKey:
+    Type: AWS::KMS::Key
+    Properties:
+      Description: !Sub "KMS key used to protect DynamoDB data in ${AWS::StackName}"
+      Enabled: true
+      EnableKeyRotation: true
+      KeyPolicy:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS: !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            Action:
+              - kms:*
+            Resource: "*"
+          - Effect: Allow
+            Principal:
+              Service: !Sub "dynamodb.amazonaws.com"
+            Action:
+              - "kms:Encrypt*"
+              - "kms:Decrypt*"
+              - "kms:ReEncrypt*"
+              - "kms:GenerateDataKey*"
+              - "kms:Describe*"
+            Resource: "*"
```

Comment:
🤔 could we tighten up the resource definition here?
This is for the Dynamo right? Could we use a `!Ref SessionsDynamoDB` here from the above session to narrow the scope a bit?

Reply:
Hey @huwd, we can't do an explicit `!Ref SessionDynamoDB` as the DynamoDB references the KMS key, so a KMS key to DynamoDB reference would cause a cyclic dependency where each resource is dependent on the creation of the other. I see Mina has defined a condition which only allows resources of ArnLike `arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:/${AWS::StackName}` but we can shift the ARN to the Resource block definition, remove the ArnLike condition, and slightly update the interpolated value to be `arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/${AWS::StackName}-SessionsDynamoDB` to match the ARN of the DynamoDB.